### PR TITLE
British Columbia Disability Credit

### LINKED ISSUE #464: British Columbia Disability Credit

### PR DESCRIPTION
**Pull request by 7qiwx**
Fixes #464

### CODE REVIEW

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount_total.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+description: British Columbia limits the disability credit to this maximum base amount.
+values:
+  2023-01-01: 14_228
+metadata:
+  unit: currency-CAD
+  period: year
+  label: British Columbia disability credit max amount
```
Comment (Yaohhhh) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount_total.yaml:7]: label: British Columbia max disability credit amount

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/younger.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+description: British Columbia provides this additional disability credit amount for younger filers.
+values:
+  2023-01-01: 5_242
+metadata:
+  unit: currency-CAD
+  period: year
+  label: British Columbia disability credit younger max amount
```
Comment (Yaohhhh) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/younger.yaml:7]: label: British Columbia max disability credit amount for the younger

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/base.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+description: British Columbia disability credit base amount.
+values:
+  2023-01-01: 8_986
+metadata:
+  unit: currency-CAD
+  label: British Columbia disability credit base
```
Comment (Yaohhhh) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/base.yaml:6]: label: British Columbia base disability credit amount

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+description: British Columbia provides an additional disability credit to filers below this age
+values:
+  2023-01-01: 18
+metadata:
+  unit: year
+  period: year
+  label: British Columbia disability credit additional amount age eligibility
```
Comment (Yaohhhh) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:7]: label: British Columbia disability credit eligible age threshold

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+description: British Columbia reduces the additional disability credit amount by the excess of childcare expenses over this threshold.  
+values:
+  2023-01-01: 3_044
```
Comment (Yaohhhh) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml:3]: Add 2022 data

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount_total.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+description: British Columbia limits the disability credit to this maximum base amount.
+values:
+  2023-01-01: 14_228
```
Comment (Yaohhhh) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount_total.yaml:3]: Add 2022 data and same as other threshold values

File: policyengine_canada/tests/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
+- name: Eligible person over 18
```
Comment (Yaohhhh) [inline policyengine_canada/tests/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit.yaml:1]: Add 2022 test scenarios.

File: policyengine_canada/tests/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_eligible.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+- name: Eligible person with disability
```
Comment (Yaohhhh) [inline policyengine_canada/tests/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_eligible.yaml:1]: Add 2022 test scenarios.

File: policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_eligible.py [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+class bc_disability_credit_eligible(Variable):
+    value_type = bool
+    entity = Person
+    label = "Eligible for the British Columbia disability tax credit"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = ProvinceCode.BC
+
+    def formula(person, period):
+        return person("is_disabled", period)
```
Comment (Yaohhhh) [inline policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_eligible.py:13]: Adding the age-eligible (<18) test to this variable can match the original policy better.

File: policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_eligible.py [hunk 10]
```diff
@@ -0,0 +1,13 @@
+from policyengine_canada.model_api import *
+
+
+class bc_disability_credit_eligible(Variable):
+    value_type = bool
+    entity = Person
+    label = "Eligible for the British Columbia disability tax credit"
```
Comment (Yaohhhh) [inline policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_eligible.py:7]: label = "British Columbia disability tax credit eligibility"

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml (hunk 5 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml:3]: yes!

File: policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_eligible.py [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+    unit = CAD
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_eligible.py:8]:
```suggestion
```

File: policyengine_canada/variables/household/person/childcare_received.py [hunk 12]
```diff
@@ -0,0 +1,9 @@
+from policyengine_canada.model_api import *
+
+
+class childcare_received(Variable):
+    value_type = float
+    entity = Person
+    label = "Childcare received"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/household/person/childcare_received.py:7]:
```suggestion
    label = "Amount of Childcare received which was claimed by another person"
```

File: policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit.py [hunk 13]
```diff
@@ -0,0 +1,30 @@
+from policyengine_canada.model_api import *
+
+
+class bc_disability_credit(Variable):
+    value_type = float
+    entity = Person
+    label = "British Columbia disability tax credit"
+    unit = CAD
+    definition_period = YEAR
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit.py:9]: add a reference to the variables - tax form is good

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  unit: currency-CAD
+  period: year
+  label: British Columbia disability credit income threshold
+  reference:
+  - title: B.C. basic personal income tax credits
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml:9]: add a tax form as well

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml:9]: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5010-d/5010-d-22e.pdf

File: policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit.py [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+        childcare_received = person("childcare_received", period)
+        additional_amount_reduction = max_(
+            0, childcare_received - p.additional_amount.income_threshold
+        )
+        additional_amount_eligibile = (
+            person("age", period) < p.additional_amount.eligible_age
+        )
+        max_additional_amount = max_(
+            0, p.additional_amount.younger - additional_amount_reduction
+        )
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit.py:23]: flip the variables to be "in order"

File: policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit.py
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit.py:1]: split the variables into the following:
bc_disability_credit_eligible - as it is 
bc_disability_credit_additional_amount_eligible - age eligibility
bc_disability_credit_additional_amount - amount calculated defined for additional eligible
bc_disability_credit - additional amount + base defined for eligible

File: changelog_entry.yaml [hunk 16]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - British Columbia Disability Credit.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - British Columbia disability credit.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]
once done please request review form @Yaohhhh

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml (hunk 4 above)
Comment (7qiwx) [reply policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:7]: It's providing additional disability credit for filers under this age, so I think 'British Columbia additional disability credit eligible age limit' describes it more accurately.

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml [hunk 17]
```diff
@@ -0,0 +1,12 @@
+description: British Columbia provides an additional disability credit to filers below this age
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:1]:
```suggestion
description: British Columbia provides an additional disability credit to filers below this age.
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:1]: rename age_limit

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+values:
+  2023-01-01: 18
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:3]: need to find 2022 values for all parameter values

Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:3]:
```suggestion
  2022-01-01: 18
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  unit: year
+  period: year
+  label: British Columbia additional disability credit eligible age limit
+  reference:
+  - title: B.C. basic personal income tax credits
+    href: https://www2.gov.bc.ca/gov/content/taxes/income-taxes/personal/credits/basic
+  - title: Income Tax Act (4.51)
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:11]:
```suggestion
  - title: British Columbia Income Tax Act (4.51)
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml [hunk 20]
```diff
@@ -0,0 +1,12 @@
+description: British Columbia provides an additional disability credit to filers below this age
+values:
+  2023-01-01: 18
+metadata:
+  unit: year
+  period: year
+  label: British Columbia additional disability credit eligible age limit
+  reference:
+  - title: B.C. basic personal income tax credits
+    href: https://www2.gov.bc.ca/gov/content/taxes/income-taxes/personal/credits/basic
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:10]: add forms for 2022 and 2023

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml [hunk 21]
```diff
@@ -0,0 +1,12 @@
+description: British Columbia provides an additional disability credit to filers below this age
+values:
+  2023-01-01: 18
+metadata:
+  unit: year
+  period: year
+  label: British Columbia additional disability credit eligible age limit
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:7]:
```suggestion
  label: British Columbia additional disability credit age limit
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml (hunk 20 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:10]: if we cant find form, add comment

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml [hunk 22]
```diff
@@ -0,0 +1,12 @@
...
+  reference:
+  - title: B.C. basic personal income tax credits
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/eligible_age.yaml:9]: give more explanatory name (year, form number etc.)

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount_total.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount_total.yaml:1]: rename to max_amount

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount_total.yaml [hunk 23]
```diff
@@ -0,0 +1,15 @@
+description: British Columbia limits the disability credit to this maximum base amount.
+values:
+  2022-01-01: 13_423
+  2023-01-01: 14_228
+metadata:
+  unit: currency-CAD
+  period: year
+  label: British Columbia max disability credit amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount_total.yaml:8]:
```suggestion
  label: British Columbia disability credit max amount
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/base.yaml [hunk 24]
```diff
@@ -0,0 +1,14 @@
+description: British Columbia disability credit base amount.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/base.yaml:1]:
```suggestion
description: British Columbia provides this disability credit base amount.
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/base.yaml [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+values:
+  2022-01-01: 8_477
+  2023-01-01: 8_986
+metadata:
+  unit: currency-CAD
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/base.yaml:6]: need period

File: policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_additional_amount.py [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+    reference = "href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5010-d/5010-d-22e.pdf#page=1"
+    defined_for = "bc_disability_credit_additional_amount_eligible"
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.bc.tax.income.credits.disability
+        childcare_received = person("childcare_received", period)
+        additional_amount_reduction = max_(
+            0, childcare_received - p.additional_amount.income_threshold
+        )
+        additional_amount = max_(
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_additional_amount.py:19]:
```suggestion
        return max_(
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/age_limit.yaml [hunk 27]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  unit: year
+  period: year
+  label: British Columbia additional disability credit age limit
+  reference:
+  - title: B.C. basic personal income tax credits 2022 and 2023 parameter values
+    href: https://www2.gov.bc.ca/gov/content/taxes/income-taxes/personal/credits/basic
+  - title: British Columbia Income Tax Act (4.51)
+    href: https://www.bclaws.gov.bc.ca/civix/document/id/complete/statreg/96215_00_multi#section4.51
+  - title: British Columbia disability amount worksheet 2022 # Did not find 2023 worksheet
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/age_limit.yaml:13]:
```suggestion
  - title: British Columbia disability amount worksheet 2022 
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml [hunk 28]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  unit: currency-CAD
+  period: year
+  label: British Columbia disability credit income threshold
+  reference:
+  - title: B.C. basic personal income tax credits 2022 and 2023 parameter values
+    href: https://www2.gov.bc.ca/gov/content/taxes/income-taxes/personal/credits/basic
+  - title: British Columbia Income Tax Act (4.51)
+    href: https://www.bclaws.gov.bc.ca/civix/document/id/complete/statreg/96215_00_multi#section4.51
+  - title: British Columbia disability amount worksheet 2022 # Did not find 2023 worksheet
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml:14]:
```suggestion
  - title: British Columbia disability amount worksheet 2022 
```

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml:14]: and others

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/younger.yaml [hunk 29]
```diff
@@ -0,0 +1,15 @@
+description: British Columbia provides this additional disability credit amount for younger filers.
+values:
+  2022-01-01: 4_946
+  2023-01-01: 5_242
+metadata:
+  unit: currency-CAD
+  period: year
+  label: British Columbia max disability credit amount for the younger
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/younger.yaml:8]:
```suggestion
  label: British Columbia max disability credit younger amount 
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount.yaml:1]: rename to cap.yaml

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount.yaml [hunk 30]
```diff
@@ -0,0 +1,15 @@
+description: British Columbia limits the disability credit to this maximum base amount.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount.yaml:1]:
```suggestion
description: British Columbia caps the disability credit at this amount.
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount.yaml [hunk 31]
```diff
@@ -0,0 +1,15 @@
...
+values:
+  2022-01-01: 13_423
+  2023-01-01: 14_228
+metadata:
+  unit: currency-CAD
+  period: year
+  label: British Columbia disability credit max amount
+  reference:
+  - title: B.C. basic personal income tax credits 2022 and 2023 parameter values
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount.yaml:10]:
```suggestion
  - title: B.C. basic personal income tax credits 
```

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/max_amount.yaml:10]: and other yaml files

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/base.yaml [hunk 32]
```diff
@@ -0,0 +1,15 @@
+description: British Columbia provides this disability credit base amount.
+values:
+  2022-01-01: 8_477
+  2023-01-01: 8_986
+metadata:
+  unit: currency-CAD
+  period: year
+  label: British Columbia base disability credit amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/base.yaml:8]:
```suggestion
  label: British Columbia disability credit base amount
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/age_limit.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/age_limit.yaml:1]: make this a single_amount parameter with the values of younger as the amount and the ages as thresholds 
0 - 18 - younger amount 
over 18 - 0

File: policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit.py [hunk 33]
```diff
@@ -0,0 +1,18 @@
...
+    unit = CAD
+    definition_period = YEAR
+    reference = "href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5010-d/5010-d-22e.pdf#page=1"
+    defined_for = "bc_disability_credit_eligible"
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.bc.tax.income.credits.disability
+        additional_amount = person(
+            "bc_disability_credit_additional_amount", period
+        )
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit.py:17]:
```suggestion
        )
        total_credit_amount = p.base + additional_amount
```

File: policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_additional_amount.py [hunk 34]
```diff
@@ -0,0 +1,21 @@
...
+    label = "British Columbia additional disability tax credit"
+    unit = CAD
+    definition_period = YEAR
+    reference = "href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5010-d/5010-d-22e.pdf#page=1"
+    defined_for = "bc_disability_credit_additional_amount_eligible"
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.bc.tax.income.credits.disability
+        childcare_received = person("childcare_received", period)
+        additional_amount_reduction = max_(
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_additional_amount.py:16]:
```suggestion
        reduced_childcare_expenses = max_(
```

File: policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_additional_amount.py [hunk 35]
```diff
@@ -0,0 +1,21 @@
...
+    entity = Person
+    label = "British Columbia additional disability tax credit"
+    unit = CAD
+    definition_period = YEAR
+    reference = "href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5010-d/5010-d-22e.pdf#page=1"
+    defined_for = "bc_disability_credit_additional_amount_eligible"
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.bc.tax.income.credits.disability
+        childcare_received = person("childcare_received", period)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/bc/tax/income/credits/bc_disability_credit/bc_disability_credit_additional_amount.py:15]:
```suggestion
        childcare_exepenses = person("childcare_received", period)
```

File: policyengine_canada/variables/household/person/childcare_received.py [hunk 36]
```diff
@@ -0,0 +1,9 @@
+from policyengine_canada.model_api import *
+
+
+class childcare_received(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/household/person/childcare_received.py:4]:
```suggestion
class childcare_expenses_claimed_by_another_person(Variable):
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/additional_amount_base.yaml [hunk 37]
```diff
@@ -0,0 +1,26 @@
+description: British Columbia provides an additional disability credit to filers by age.
+brackets: 
+  - amount:
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/additional_amount_base.yaml:3]: start with the threshold in the breakdown

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/additional_amount_base.yaml [hunk 38]
```diff
@@ -0,0 +1,26 @@
...
+  - amount: 
+      2022-01-01: 0
+    threshold:
+      2022-01-01: 18
+metadata:
+  type: single_amount
+  threshold_unit: year
+  amount_unit: currency-CAD
+  period: year
+  name: bc_additional_disability_credit
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/additional_amount_base.yaml:17]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/additional_amount_base.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/additional_amount_base.yaml:1]: rename to base.yaml

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/additional_amount_base.yaml [hunk 39]
```diff
@@ -0,0 +1,26 @@
+description: British Columbia provides an additional disability credit to filers by age.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/additional_amount_base.yaml:1]:
```suggestion
description: British Columbia provides an additional disability credit of this amount, based on age.
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/additional_amount_base.yaml [hunk 40]
```diff
@@ -0,0 +1,26 @@
...
+      2022-01-01: 0
+    threshold:
+      2022-01-01: 18
+metadata:
+  type: single_amount
+  threshold_unit: year
+  amount_unit: currency-CAD
+  period: year
+  name: bc_additional_disability_credit
+  label: British Columbia additional disability credit
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/additional_amount_base.yaml:18]:
```suggestion
  label: British Columbia disability credit additional amount base
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/cap.yaml [hunk 41]
```diff
@@ -0,0 +1,15 @@
+description: British Columbia caps the disability credit at this amount.
+values:
+  2022-01-01: 13_423
+  2023-01-01: 14_228
+metadata:
+  unit: currency-CAD
+  period: year
+  label: British Columbia disability credit max amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/cap.yaml:8]:
```suggestion
  label: British Columbia disability credit cap
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml:1]: rename to childcare_expense_threshold

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml [hunk 42]
```diff
@@ -0,0 +1,15 @@
+description: British Columbia reduces the additional disability credit amount by the excess of childcare expenses over this threshold.  
+values:
+  2022-01-01: 2_871
+  2023-01-01: 3_044
+metadata:
+  unit: currency-CAD
+  period: year
+  label: British Columbia disability credit income threshold
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/income_threshold.yaml:8]:
```suggestion
  label: British Columbia disability credit additional amount childcare expense threshold
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/variables/household/person/childcare_expenses_claimed_by_another_person.py [hunk 43]
```diff
@@ -0,0 +1,9 @@
+from policyengine_canada.model_api import *
+
+
+class childcare_expenses_claimed_by_another_person(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/household/person/childcare_expenses_claimed_by_another_person.py:4]: @MaxGhenis is there a better way to represent this variable?

File: policyengine_canada/variables/household/person/childcare_expenses_claimed_by_another_person.py [hunk 44]
```diff
@@ -0,0 +1,9 @@
...
+    value_type = float
+    entity = Person
+    label = "Amount of Childcare received which was claimed by another person"
+    unit = CAD
+    definition_period = YEAR
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/household/person/childcare_expenses_claimed_by_another_person.py:9]: from policyengine_canada.model_api import *


class childcare_received(Variable):
    value_type = float
    entity = Person
    label = "Amount of Childcare received which was claimed by another person"
    unit = CAD
    definition_period = YEAR

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/cap.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/additional_amount/cap.yaml:1]: move this out of the additional amount folder as this applied to the whole credit

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/non_refundable.yaml [hunk 45]
```diff
@@ -3,6 +3,7 @@ values:
   2022-01-01:
     - bc_tax_reduction_credit
     - bc_age_credit
+    - bc_disability_credit
 
 metadata:
   unit: program
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/non_refundable.yaml:9]: could you add 
label
period 
references

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/non_refundable.yaml [hunk 46]
```diff
@@ -3,6 +3,7 @@ values:
   2022-01-01:
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/non_refundable.yaml:3]: description: British Columbia provides the following non-refundable tax credits.

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/cap.yaml [hunk 47]
```diff
@@ -0,0 +1,15 @@
+description: British Columbia caps the disability credit at this amount.
+values:
+  2022-01-01: 13_423
+  2023-01-01: 14_228
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/disability/cap.yaml:4]: isn't this just the sum of the base and additional amount? if so i don't think we need it

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**